python-package/lightgbm/callback.py

```diff
@@ -214,7 +215,25 @@ def _callback(env):
             _init(env)
         if not enabled[0]:
             return
+        if first_metric_only:
+            eval_metric = None
+            for m in ['metric', 'metrics', 'metric_types']:
```

Comment:
`metric_alias`? Please avoid using one-letter names, except it's something conventional like `i`/`j` for numerical loops.